xdsl/dialects/riscv_func.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from typing import Annotated
+
+from xdsl.ir import Operation, SSAValue, Dialect
+from xdsl.utils.exceptions import VerifyException
+
+from xdsl.irdl import (
+    IRDLOperation,
+    OptOpResult,
+    VarOperand,
+    irdl_op_definition,
+    OpAttr,
+)
+from xdsl.dialects.builtin import AnyIntegerAttr, IntegerAttr, IntegerType
+from xdsl.dialects import riscv
+
+
+@irdl_op_definition
+class SyscallOp(IRDLOperation):
+    name = "riscv_func.syscall"
+    args: Annotated[VarOperand, riscv.RegisterType]
+    syscall_num: OpAttr[IntegerAttr[IntegerType]]
+    result: Annotated[OptOpResult, riscv.RegisterType]
+    """
+    Some syscalls return values by putting them into a0. If result is not None, then the
+    contents of a0 will be moved to its register.
+    """
```

Comment:
afaik, it is not uncommon for syscalls to return values in multiple registers (a way around the `-errno` issue).

Reply:
In that case this comment is just misleading, I removed it